lib/feature_map/private/additional_metrics_file.rb

```diff
@@ -0,0 +1,164 @@
+          health = health_score_for(cyclomatic_complexity, encapsulation, test_coverage, health_config)
+
+          feature_additional_metrics[feature_name] = {
+            'cyclomatic_complexity' => cyclomatic_complexity,
+            'encapsulation' => encapsulation,
+            'feature_size' => feature_size,
+            'test_coverage' => test_coverage,
+            'health' => health
+          }
+        end
+
+        { FEATURES_KEY => feature_additional_metrics }
+      end
+
+      sig { returns(FeaturesContent) }
+      def self.load_features!
+        metrics_content = YAML.load_file(path)
+
+        return metrics_content[FEATURES_KEY] if metrics_content.is_a?(Hash) && metrics_content[FEATURES_KEY]
+
+        raise FileContentError, "Unexpected content found in #{path}. Use `bin/featuremap validate` to regenerate it and try again."
+      rescue Psych::SyntaxError => e
+        raise FileContentError, "Invalid YAML content found at #{path}. Error: #{e.message} Use `bin/featuremap validate` to generate it and try again."
+      rescue Errno::ENOENT
+        raise FileContentError, "No feature metrics file found at #{path}. Use `bin/featuremap validate` to generate it and try again."
+      end
+
+      sig { params(collection: T::Array[T.any(Integer, Float)], score: T.any(Integer, Float)).returns({ 'percentile' => Float, 'percent_of_max' => Integer, 'score' => T.any(Integer, Float) }) }
+      def self.calculate(collection, score)
+        max = collection.max || 0
+        percentile = percentile_of(collection, score)
+        percent_of_max = max.zero? ? 0 : ((score.to_f / max) * 100).round.to_i
+
+        { 'percentile' => percentile, 'percent_of_max' => percent_of_max, 'score' => score }
+      end
+
+      sig { params(arr: T::Array[T.any(Integer, Float)], val: T.any(Integer, Float)).returns(Float) }
+      def self.percentile_of(arr, val)
+        return 0.0 if arr.empty?
+
+        ensure_array_of_floats = arr.map(&:to_f)
+        ensure_float_value = val.to_f
+
+        below_or_equal_count = ensure_array_of_floats.reduce(0) do |acc, v|
+          if v < ensure_float_value
+            acc + 1
+          elsif v == ensure_float_value
+            acc + 0.5
+          else
+            acc
+          end
+        end
+
+        ((100 * below_or_equal_count) / ensure_array_of_floats.length).to_f
+      end
+
+      sig { params(awardable_points: Integer, score: T.any(Float, Integer), score_threshold: Integer, percent_of_max: T.nilable(T.any(Integer, Float)), percent_of_max_threshold: T.nilable(T.any(Integer, Float))).returns({ 'awardable_points' => Integer, 'health_score' => T.any(Float, Integer), 'close_to_maximum_score' => T::Boolean, 'exceeds_score_threshold' => T::Boolean }) }
+      def self.health_score_component(awardable_points, score, score_threshold, percent_of_max = 0, percent_of_max_threshold = 100)
+        close_to_maximum_score = T.must(percent_of_max) >= T.must(percent_of_max_threshold)
```

Comment:
The fact that `percent_of_max` and `percent_of_max_threshold ` are defined in the signature for this method as `T.nilable(...)` and that you assert them as `T.must(...)` here, leads me to believe that there is a nil check bug lurking in the code path that results in a nil value being passed in as one or both of these values? Is this a reasonable concern or is there a reason this is not a potential issue in this case.

This kinds of type safety bugs seem like the type of issue that Sorbet would ideally highlight for us and help us avoid.

Reply:
Spoke about this offline and decided to punt on this until we get clarity around whether or not we plan to keep Sorbet around.